sdk/core/azure-core/src/main/java/com/azure/core/http/policy/AzureSasCredentialPolicy.java

```diff
@@ -29,18 +29,12 @@
      */
     public AzureSasCredentialPolicy(AzureSasCredential credential) {
         Objects.requireNonNull(credential, "'credential' cannot be null.");
-
         this.credential = credential;
     }
 
     @Override
     public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
         HttpRequest httpRequest = context.getHttpRequest();
-        if ("http".equals(httpRequest.getUrl().getProtocol())) {
-            return Mono.error(new IllegalStateException(
-                "Shared access signature credentials require HTTPS to prevent leaking the shared access signature."));
-        }
```

Comment:
Even though the portal supports creating SAS tokens that use HTTP, I am inclined to only support HTTPS to begin with as we would ideally want to minimize HTTP usage. If there's a customer ask and switching to HTTPS is not an option for them, then we can remove this check instead of proactively removing it. Thoughts?

Reply:
I agree that we should promote HTTPS. However, there's a complication. We already support SAS over HTTP [here](https://github.com/Azure/azure-sdk-for-java/blob/99fed03e150931c678e30a0db5f3ca7d7fc05dee/sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobClientBuilder.java#L206-L219). I was thinking about replacing that implementation with new AzureSasCredential and adding overloads that also take AzureSasCredential to allow rotation. However if we do that the sasToken(String) would start throwing if it's not https... Other option is to keep sasToken(String) as is and deprecate them. Thoughts on that?

Reply:
Discussed offline. Will add constructor overload to customize that behavior.